src/Language/PureScript/Sugar/TypeClasses/Deriving.hs

```diff
@@ -694,11 +697,14 @@ mkVarMn mn = Var . Qualified mn
 mkVar :: Ident -> Expr
 mkVar = mkVarMn Nothing
 
+labelToIdent :: Label -> Ident
+labelToIdent (Label l) = Ident $ T.pack $ codePoints l
```

Comment:
Similarly to my review comments in the JS pretty-printer, I worry that this is function is partial in some sense. In particular I think there's a risk of it introducing unicode replacement characters in derived `Generic` instances for record types with awkward labels. I think we should probably add a test for deriving `Generic` with records with awkward labels, i.e. labels which are not valid JS identifiers and also labels which are not valid UTF-16, if we don't already have one.